hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java

```diff
@@ -0,0 +1,146 @@
+import static org.apache.hadoop.ozone.snapshot.SnapshotDiffResponse.JobStatus.IN_PROGRESS;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests snapshot in OM HA setup.
+ */
+public class TestOzoneManagerHASnapshot extends TestOzoneManagerHA {
+
+  // Test snapshot diff when OM restarts in HA OM env.
+  @Test
+  public void testSnapshotDiffWhenOmLeaderRestart()
+      throws Exception {
+    OzoneBucket ozoneBucket = setupBucket();
+    String volumeName = ozoneBucket.getVolumeName();
+    String bucketName = ozoneBucket.getName();
+    String snapshot1 = "snap-" + RandomStringUtils.randomNumeric(5);
+    String snapshot2 = "snap-" + RandomStringUtils.randomNumeric(5);
+
+    createKey(ozoneBucket);
+    getObjectStore().createSnapshot(volumeName, bucketName, snapshot1);
+
+    for (int i = 0; i < 100; i++) {
+      createKey(ozoneBucket);
+    }
+
+    getObjectStore().createSnapshot(volumeName, bucketName, snapshot2);
+
+    SnapshotDiffResponse response =
+        getObjectStore().snapshotDiff(volumeName, bucketName,
+            snapshot1, snapshot2, null, 0, false);
+
+    assertEquals(IN_PROGRESS, response.getJobStatus());
+
+    String oldLeader = getCluster().getOMLeader().getOMNodeId();
+
+    OzoneManager omLeader = getCluster().getOMLeader();
+    getCluster().shutdownOzoneManager(omLeader);
+    getCluster().restartOzoneManager(omLeader, true);
+
+    await().atMost(Duration.ofSeconds(120)).
+        until(() -> getCluster().getOMLeader() != null);
+
+    String newLeader = getCluster().getOMLeader().getOMNodeId();
+
+    if (Objects.equals(oldLeader, newLeader)) {
+      // If old leader becomes leader again. Wait for some time to snapshot diff
+      // job finish because OM will load IN_PROGRESS on the startup.
+      Thread.sleep(1000L);
+      response =
+          getObjectStore().snapshotDiff(volumeName, bucketName, snapshot1,
+              snapshot2, null, 0, false);
+      assertEquals(DONE, response.getJobStatus());
+      assertEquals(100, response.getSnapshotDiffReport().getDiffList().size());
+    } else {
+      // If new leader is different from old leader. SnapDiff request will be
+      // new to OM, and job status should be IN_PROGRESS.
+      response =
+          getObjectStore().snapshotDiff(volumeName, bucketName, snapshot1,
+              snapshot2, null, 0, false);
+      assertEquals(IN_PROGRESS, response.getJobStatus());
+      while (true) {
+        response =
+            getObjectStore().snapshotDiff(volumeName, bucketName, snapshot1,
+                snapshot2, null, 0, false);
+        if (DONE == response.getJobStatus()) {
+          assertEquals(100,
+              response.getSnapshotDiffReport().getDiffList().size());
+          break;
+        }
+        Thread.sleep(response.getWaitTimeInMs());
+      }
+    }
+  }
+
+  @Test
+  public void testUniqueSnapshotId() throws Exception {
```

Comment:
Maybe rename the test since it verifies if the same snapshot's UUID is consistent among all OMs?

```suggestion
  public void testSnapshotIdConsistency() throws Exception {
```